bikeshed/update/manifest.py

```diff
@@ -143,28 +145,54 @@ def updateByManifest(path, dryRun=False):
         lastMsgTime = time.time()
         if newPaths:
             say("Updating {0} file{1}...", len(newPaths), "s" if len(newPaths) > 1 else "")
+
+        updateQueue = queue.Queue()
+        success = True
+        # This worker will download from remote, and update success if it fails.
+        def worker():
+            # Needed to update status and progress.
+            nonlocal success
+            nonlocal lastMsgTime
+            while True:
+                (i, filePath) = updateQueue.get()
+                remotePath = ghPrefix + filePath
+                localPath = localizePath(path, filePath)
+                try:
+                    newFile = requests.get(remotePath).text
+                except Exception as e:
+                    warn("Couldn't download file '{0}'.\n{1}", remotePath, e)
+                    success = False
+                try:
+                    dirPath = os.path.dirname(localPath)
+                    if not os.path.exists(dirPath):
+                        os.makedirs(dirPath)
+                    with io.open(localPath, 'w', encoding="utf-8") as fh:
+                        fh.write(newFile)
+                except Exception as e:
+                    warn("Couldn't save file '{0}'.\n{1}", localPath, e)
+                    success = False
+
+                currFileTime = time.time()
+                if (currFileTime - lastMsgTime) >= messageDelta:
+                    say("Updated {0}/{1}...", i+1, len(newPaths))
+                    lastMsgTime = currFileTime
+                updateQueue.task_done()
+
+        # Enqueue all the files to update. i is used for printing progress.
         for i,filePath in enumerate(newPaths):
-            remotePath = ghPrefix + filePath
-            localPath = localizePath(path, filePath)
-            try:
-                newFile = requests.get(remotePath).text
-            except Exception as e:
-                warn("Couldn't download file '{0}'.\n{1}", remotePath, e)
-                return False
-            try:
-                dirPath = os.path.dirname(localPath)
-                if not os.path.exists(dirPath):
-                    os.makedirs(dirPath)
-                with io.open(localPath, 'w', encoding="utf-8") as fh:
-                    fh.write(newFile)
-            except Exception as e:
-                warn("Couldn't save file '{0}'.\n{1}", localPath, e)
-                return False
-
-            currFileTime = time.time()
-            if (currFileTime - lastMsgTime) >= messageDelta:
-                say("Updated {0}/{1}...", i+1, len(newPaths))
-                lastMsgTime = currFileTime
+            updateQueue.put((i, filePath))
+
+        # Create as many threads as we can.
+        for i in range(len(os.sched_getaffinity(0))):
+            t = threading.Thread(target=worker, daemon=True).start()
```

Comment:
https://docs.aiohttp.org/en/stable/ suggests that we can avoid threads entirely by using a different fetching library.

Reply:
Can you point to where in that page it mentions avoiding threads?
Also I'm hesitant to pull in an entire library just to make this parallel, but if @tabatkins is okay with it, I can do that!

Reply:
These threads don't seem too complicated, so I'm fine with using them straight for now.

Reply:
https://docs.aiohttp.org/en/stable/glossary.html#term-asyncio mentions that it's single-threaded, which reduces how many race conditions you need to handle.